grammar.js

```diff
@@ -1027,7 +1027,7 @@ const rules = {
 
   xhp_comment: $ => token(seq('<!--', /(-?>)?([^>]|[^-]>|[^-]->)*/, '-->')),
 
-  xhp_string: $ => /[^<{]+/,
+  xhp_string: $ => token(prec(1, /[^<{]+/)),
```

Comment:
Oh nice! Didn't expect the solution to be as straight forward. Much simpler than what I was thinking!